core/utils.ts

```diff
@@ -263,3 +255,44 @@ export function flatten<T>(nestedArray: T[][]) {
     return previousValue.concat(currentValue);
   }, []);
 }
+
+export function matchFQName(
```

Comment:
if i understand correctly, this code can be simplified a fair bit:

- this is only called by `matchPatterns`
- `matchPatterns` is called in two places; one for determining what dependencies should be included, and one for determining what actions can be included
- since dependencies are fully-qualified now, we can stop using `matchPatterns` in the first case - we simply need to find the intersection of `action.dependencies` and `allActionNames`, something like `action.dependencies.filter(dependency => allActionNames.includes(dependency))`
- in the case of actions, we only care about (a) if the requested action name includes a `.`, then we should be able to fully-match it to an action name in the graph, or (b) if the requested action name does not include a `.`, then we should only match against the last segment of each action name:

```
export function matchPatterns(patterns: string[], values: string[]) {
  const fQActs: string[] = [];
  patterns.forEach(pat => {
    if (pat.includes(".")) {
      if (values.includes(pat)) {
        fQActs.push(pat);
      }
    } else {
      const matchingActions = values.filter(value => pat === value.split(".").slice(-1)[0]);
      if (matchingActions.length === 0) {
        return;
      }
      if (matchingActions.length > 1) {
        throw new Error(...);
      }
      fQActs.push(matchingActions[0]);
    }
  });
  return fQActs;
}
```

Reply:
Done!